### PR TITLE
Add breaking changes for WPF

### LINKED ISSUE #43282: [Breaking change]: WPF for .NET Core 3.0 altered how the text editor controls set data objects for drag operations
### Description

.NET Core 3.0 introduced a change for how the text editor controls would create a `System.Windows.DataObject` when grabbing text to drag to another control. The code disabled autoconversion, which caused the operation to not convert the data to `DataFormats.StringFormat` and instead keep it as `DataFormats.Text` or `DataFormats.UnicodeText`. This was a change 

### Version

Other (please put exact version in description textbox)

### Previous behavior

The data type on `System.Windows.DataObject` when dragging text from a text editor control was `DataFormats.StringFormat`.

### New behavior

The data type on `System.Windows.DataObject` when dragging text from a text editor control is `DataFormats.Text` or `DataFormats.UnicodeText`.

### Type of breaking change

- [ ] **Binary incompatible**: Existing binaries might encounter a breaking change in behavior, such as failure to load or execute, and if so, require recompilation.
- [ ] **Source incompatible**: When recompiled using the new SDK or component or to target the new runtime, existing source code might require source changes to compile successfully.
- [x] **Behavioral change**: Existing binaries might behave differently at run time.

### Reason for change

Unintentional change.

### Recommended action

This change was reverted in .NET 7. Upgrade to .NET 7 or later.

### Feature area

Windows Presentation Foundation (WPF)

### Affected APIs

_No response_

---
[Associated WorkItem - 340207](https://dev.azure.com/msft-skilling/Content/_workitems/edit/340207)

### LINKED ISSUE #43285: [Breaking change]: WPF for .NET 7 fixes a bug introduced in .NET Core 3.0 related to how the text editor controls set data objects for drag operations
### Description

.NET 7 fixes a regression introduced in .NET Core 3.0 related to how drag operations set data when dragging text from a text editor control. This restores behavior to match .NET Framework's behavior.

When dragging text from a text editor, the text is set as `DataFormats.Text` or `DataFormats.UnicodeText` on the `System.Windows.DataObject` object, but with autoconversion turned on. This converts the data to the `DataFormats.StringFormat` format on the `DataObject`.

### Version

Other (please put exact version in description textbox)

### Previous behavior

The data type on `System.Windows.DataObject` when dragging text from a text editor control is `DataFormats.Text` or `DataFormats.UnicodeText`.

### New behavior

The data type on `System.Windows.DataObject` when dragging text from a text editor control is `DataFormats.StringFormat`.

### Type of breaking change

- [ ] **Binary incompatible**: Existing binaries might encounter a breaking change in behavior, such as failure to load or execute, and if so, require recompilation.
- [ ] **Source incompatible**: When recompiled using the new SDK or component or to target the new runtime, existing source code might require source changes to compile successfully.
- [x] **Behavioral change**: Existing binaries might behave differently at run time.

### Reason for change

Match the behavior of .NET Framework.

### Recommended action

Upgrade older projects to the latest version of .NET to restore the correct behavior.

### Feature area

Windows Presentation Foundation (WPF)

### Affected APIs

_No response_

---
[Associated WorkItem - 340206](https://dev.azure.com/msft-skilling/Content/_workitems/edit/340206)

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

Document two old breaking changes.

@gewarren I improvised on the designs to match the latest way of documenting these I could see.

Fixes #43282
Fixes #43285

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/3.0.md](https://github.com/dotnet/docs/blob/183944da726f93670a3ddefbcb016644ecc62c86/docs/core/compatibility/3.0.md) | [docs/core/compatibility/3.0](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/3.0?branch=pr-en-us-43309) |
| [docs/core/compatibility/7.0.md](https://github.com/dotnet/docs/blob/183944da726f93670a3ddefbcb016644ecc62c86/docs/core/compatibility/7.0.md) | [docs/core/compatibility/7.0](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/7.0?branch=pr-en-us-43309) |
| [docs/core/compatibility/wpf.md](https://github.com/dotnet/docs/blob/183944da726f93670a3ddefbcb016644ecc62c86/docs/core/compatibility/wpf.md) | [docs/core/compatibility/wpf](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/wpf?branch=pr-en-us-43309) |
| [docs/core/compatibility/wpf/7.0/drag-and-drop.md](https://github.com/dotnet/docs/blob/183944da726f93670a3ddefbcb016644ecc62c86/docs/core/compatibility/wpf/7.0/drag-and-drop.md) | [docs/core/compatibility/wpf/7.0/drag-and-drop](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/wpf/7.0/drag-and-drop?branch=pr-en-us-43309) |

### CODE REVIEW

File: docs/core/compatibility/wpf/3.0/drag-and-drop.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: "Breaking change: 'Text drag-and-drop operations'"
+description: Learn about a breaking change in Windows Presentation Foundation (WPF) and .NET Core 3.0. Drag-and-drop behavior changed when dragging text from a text editor control.
```
Comment (gewarren) [inline docs/core/compatibility/wpf/3.0/drag-and-drop.md:3]:
```suggestion
description: Learn about a breaking change in Windows Presentation Foundation (WPF) in .NET Core 3.0. Drag-and-drop behavior changed when dragging text from a text editor control.
```

File: docs/core/compatibility/wpf/3.0/drag-and-drop.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ms.date: 11/4/2024
+ai-usage: ai-assisted
+---
+
+# .NET Core 3.0 altered drag-and-drop behavior on text editors
```
Comment (gewarren) [inline docs/core/compatibility/wpf/3.0/drag-and-drop.md:8]:
```suggestion
# Altered drag-and-drop behavior on text editors
```

File: docs/core/compatibility/wpf/7.0/drag-and-drop.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
+title: "Breaking change: 'Revert behavior with text drag-and-drop operations'"
+description: Learn about a breaking change in Windows Presentation Foundation (WPF) and .NET 7. Drag-and-drop behavior was reverted to .NET Framework behavior behavior when dragging text from a text editor control.
```
Comment (gewarren) [inline docs/core/compatibility/wpf/7.0/drag-and-drop.md:3]:
```suggestion
description: Learn about a breaking change in Windows Presentation Foundation (WPF) and .NET 7. Drag-and-drop behavior was reverted to .NET Framework behavior when dragging text from a text editor control.
```

File: docs/core/compatibility/wpf/7.0/drag-and-drop.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ms.date: 11/4/2024
+ai-usage: ai-assisted
+---
+
+# .NET 7 restored drag-and-drop operations behavior on text editors
```
Comment (gewarren) [inline docs/core/compatibility/wpf/7.0/drag-and-drop.md:8]:
```suggestion
# Drag-and-drop operations in text editors
```

File: docs/core/compatibility/wpf/7.0/drag-and-drop.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+.NET 7 addresses a regression introduced in .NET Core 3.0 concerning drag operations in text editor controls. This update restores the behavior to match that of the .NET Framework, ensuring consistency in how data is set during drag operations.
```
Comment (gewarren) [inline docs/core/compatibility/wpf/7.0/drag-and-drop.md:10]:
```suggestion
.NET 7 addresses a regression introduced in .NET Core 3.0 concerning drag operations in text editor controls. This update restores the behavior to match that of .NET Framework, ensuring consistency in how data is set during drag operations.
```

File: docs/core/compatibility/wpf/7.0/drag-and-drop.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+## Version introduced
+
+.NET 7
+
+## Previous behavior
+
+The data type on <xref:System.Windows.DataObject?displayProperty=fullName> when dragging text from a text editor control is <xref:System.Windows.DataFormats.Text?displayProperty=nameWithType> or <xref:System.Windows.DataFormats.UnicodeText?displayProperty=nameWithType>.
```
Comment (gewarren) [inline docs/core/compatibility/wpf/7.0/drag-and-drop.md:18]:
```suggestion
In .NET Core 3.0 through .NET 6, the data type on <xref:System.Windows.DataObject?displayProperty=fullName> when dragging text from a text editor control was <xref:System.Windows.DataFormats.Text?displayProperty=nameWithType> or <xref:System.Windows.DataFormats.UnicodeText?displayProperty=nameWithType>.
```

File: docs/core/compatibility/wpf/7.0/drag-and-drop.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+## New behavior
+
+The data type on <xref:System.Windows.DataObject?displayProperty=fullName> when dragging text from a text editor control is <xref:System.Windows.DataFormats.StringFormat?displayProperty=nameWithType>.
```
Comment (gewarren) [inline docs/core/compatibility/wpf/7.0/drag-and-drop.md:22]:
```suggestion
Starting in .NET 7, the data type on <xref:System.Windows.DataObject?displayProperty=fullName> when dragging text from a text editor control is <xref:System.Windows.DataFormats.StringFormat?displayProperty=nameWithType>.
```

File: docs/core/compatibility/wpf/7.0/drag-and-drop.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+## Type of breaking change
+
+This change is a [behavioral change](../../categories.md#behavioral-change).
+
+## Reason for change
+
+The change is made to match the behavior of the .NET Framework, ensuring consistency and expected behavior across different .NET versions.
```
Comment (gewarren) [inline docs/core/compatibility/wpf/7.0/drag-and-drop.md:30]:
```suggestion
The change was made to revert an unintentional change in .NET 3.0 and match the behavior of .NET Framework.
```

File: docs/core/compatibility/toc.yml
Comment (gewarren) [inline docs/core/compatibility/toc.yml:1]: The .NET 7 entry should also go in the "by version" part of the TOC. So add a new WPF section for .NET 7 after the WinForms section on line 509.

File: docs/core/compatibility/toc.yml [hunk 9]
```diff
@@ -2042,12 +2042,20 @@ items:
...
           - name: .NET 5
             items:
               - name: OutputType set to WinExe
                 href: sdk/5.0/automatically-infer-winexe-output-type.md
               - name: WPF apps use Microsoft.NET.Sdk
                 href: sdk/5.0/sdk-and-target-framework-change.md
+          - name: .NET Core 3.0
+            items:
+              - name: "Text drag-and-drop operations"
+                href: wpf/3.0/drag-and-drop.md
```
Comment (gewarren) [inline docs/core/compatibility/toc.yml:2058]:
```suggestion
```

File: docs/core/compatibility/wpf/3.0/drag-and-drop.md
Comment (gewarren) [inline docs/core/compatibility/wpf/3.0/drag-and-drop.md:1]: This should be an include file to match the other breaking changes for .NET Core 3.0.

**Comment by adegeo** [comment]
Ouch! Lots of little fixes :) Thank you, I'll get on them!

File: includes/core-changes/wpf/3.0/drag-and-drop.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+This change is a [behavioral change](../../../../docs/core/compatibility/categories.md).
+
+#### Reason for change
+
+The change was unintentional.
+
+#### Recommended action
+
+This change was reverted in .NET 7. Upgrade to .NET 7 or later.
```
Comment (gewarren) [inline includes/core-changes/wpf/3.0/drag-and-drop.md:39]:
```suggestion
This change was [reverted in .NET 7](/dotnet/core/compatibility/wpf/7.0/drag-and-drop). Upgrade to .NET 7 or later.
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**